### PR TITLE
Sum of squares of first N natural numbers

### LINKED ISSUE #1273: Sum of squares of first N natural numbers
In C, C++, Java, Python and Go

### PR DESCRIPTION
**Pull request by singh-shreya6**
Fixes #1273 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Added program for sum of squares of first N natural numbers

#### Languages Used:
- C
- C++
- Java
- Python
- Go

#### Files Added:
- Sum_Of_Squares.c
- Sum_Of_Squares.cpp
- Sum_Of_Squares.java
- Sum_Of_Squares.py
- Sum_Of_Squares.go

### CODE REVIEW

File: Sum_Of_Squares/Sum_Of_Squares.cpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    ((N) * (N + 1) * (2 * N + 1)) / 6
+*/
+
+#include <bits/stdc++.h>
+using namespace std;
+
+int main()
+{
+    int N;
+    cout << "Enter the value of N: " << endl;
```
Comment (marwahmanbir) [inline Sum_Of_Squares/Sum_Of_Squares.cpp]: Please remove spaces around ``<<`` and ``>>`` in cout and cin statements as these are not required and the compiler simply ignores unnecessary spaces.

File: Sum_Of_Squares/Sum_Of_Squares.c [hunk 2]
```diff
@@ -0,0 +1,31 @@
+/*
+    The sum of squares of first N natural numbers is given
+    by the equation ((N) * (N + 1) * (2 * N + 1)) / 6 i.e.
+    1^2 + 2^2 + 3^2 + 4^2 + 5^2.........N^2 sums upto
+    ((N) * (N + 1) * (2 * N + 1)) / 6
+*/
+
+#include <stdio.h>
```
Comment (marwahmanbir) [inline Sum_Of_Squares/Sum_Of_Squares.c]: Remove space after include

File: Sum_Of_Squares/Sum_Of_Squares.cpp [hunk 3]
```diff
@@ -0,0 +1,32 @@
+/*
+    The sum of squares of first N natural numbers is given
+    by the equation ((N) * (N + 1) * (2 * N + 1)) / 6 i.e.
+    1^2 + 2^2 + 3^2 + 4^2 + 5^2.........N^2 sums upto
+    ((N) * (N + 1) * (2 * N + 1)) / 6
+*/
+
+#include <bits/stdc++.h>
```
Comment (marwahmanbir) [inline Sum_Of_Squares/Sum_Of_Squares.cpp]: Remove space after include

**Comment by singh-shreya6** [comment]
Done @marwahmanbir . Please review.

**Pull request merged**